### PR TITLE
2.0 preview

### PR DESCRIPTION
**Pull request by valdrinkoshi**
- ⚠️  breaking change ⚠️ `paper-dialog-shared-styles` changed `h2` and `.buttons` selectors to match how [`::slotted` works](https://developers.google.com/web/fundamentals/primers/shadowdom/?hl=en#stylinglightdom) (compound selector will select only top level nodes) 
- added a `suite` wrapper to keep `property` scoped also inside the tests (was causing duplicate tests)
- IIFE and use strict
- remove `paper-dialog-common.css`

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -20,21 +20,43 @@
...
+    "paper-item": "PolymerElements/paper-item#2.0-preview",
+    "paper-listbox": "PolymerElements/paper-listbox#2.0-preview",
+    "web-component-tester": "^6.0.0-prerelease.6",
+    "webcomponentsjs": "webcomponents/webcomponentsjs#^1.0.0-rc.1"
+  },
+  "variants": {
+    "1.x": {
+      "dependencies": {
+        "iron-overlay-behavior": "PolymerElements/iron-overlay-behavior#^1.0.0",
+        "paper-styles": "PolymerElements/paper-styles#^1.0.0",
```
Comment (bicknellr) [inline bower.json]: this was previously `^1.1.0`

File: bower.json [hunk 2]
```diff
@@ -20,21 +20,43 @@
...
+      "devDependencies": {
+        "iron-component-page": "PolymerElements/iron-component-page#^1.0.0",
+        "iron-demo-helpers": "PolymerElements/iron-demo-helpers#^1.0.0",
+        "iron-test-helpers": "PolymerElements/iron-test-helpers#^1.0.0",
+        "paper-button": "PolymerElements/paper-button#^1.0.0",
+        "paper-dialog-scrollable": "PolymerElements/paper-dialog-scrollable#^1.0.0",
+        "paper-dropdown-menu": "PolymerElements/paper-dropdown-menu#^1.0.0",
+        "paper-icon-button": "PolymerElements/paper-icon-button#^1.0.0",
+        "paper-item": "PolymerElements/paper-item#^1.0.0",
+        "paper-listbox": "PolymerElements/paper-listbox#^1.0.0",
```
Comment (bicknellr) [inline bower.json]: this was `^1.1.0` before.

**Comment by valdrinkoshi** [comment]
The demo wasn't working on the main variant because i forgot to assign the right slot name to the distributed content - good catch!
This is now fixed in the last commit 👌

File: demo/index.html [hunk 3]
```diff
@@ -50,14 +50,14 @@
       <paper-button raised onclick="modalAlert.toggle()">modal alert</paper-button>
       <simple-dialog id="modalAlert" modal role="alertdialog">
         <h2>Alert</h2>
-        <paper-dropdown-menu label="Draft to discard">
-          <paper-listbox class="dropdown-content">
+        <paper-dropdown-menu-light label="Draft to discard">
+          <paper-listbox class="dropdown-content" slot="dropdown-content">
```
Comment (bicknellr) [inline demo/index.html]: Could you also leave a note here about why this needs both `class="dropdown-content"` and `slot="dropdown-content"`?

Comment (valdrinkoshi) [inline demo/index.html]: Done 👌
